src/uct/tcp/tcp_sockcm.c

```diff
@@ -36,18 +37,73 @@ static uct_cm_ops_t uct_tcp_sockcm_ops = {
     .listener_create  = UCS_CLASS_NEW_FUNC_NAME(uct_tcp_listener_t),
     .listener_reject  = uct_tcp_listener_reject,
     .listener_query   = uct_tcp_listener_query,
-    .listener_destroy = UCS_CLASS_DELETE_FUNC_NAME(uct_tcp_listener_t)
+    .listener_destroy = UCS_CLASS_DELETE_FUNC_NAME(uct_tcp_listener_t),
+    .ep_create        = UCS_CLASS_NEW_FUNC_NAME(uct_tcp_sockcm_ep_t)
 };
 
+static void uct_tcp_close_sa_arg(uct_tcp_sa_arg_t *sa_arg_ctx)
+{
+    ucs_list_del(&sa_arg_ctx->list);
+    ucs_async_remove_handler(sa_arg_ctx->fd, 1);
+    close(sa_arg_ctx->fd);
+    ucs_free(sa_arg_ctx);
+}
+
+static int uct_tcp_is_fd_connected(int fd)
```

Comment:
why need this wrapper?
are you going to use it not only once? if not, i would do it directly or rename to say that we do some error handling (i.e. modify async handler)
wdyt?

Reply:
yes, i will use it more than once in the next PRs.
also, i think it's better to have the async_handler function short...